### ISSUE TITLE
Using a smartwatch as navigation display

### ISSUE THREAD

**Issue by arnowelzel** [osmandapp/OsmAnd#19615]
### Describe the idea (required)

I am aware that this overlaps with #14406.

I've been using a Samsung Gear S3 with a companion app which forwards navigation information from Google Maps to the smartwatch: direction arrow where to drive to (straight ahead, left, right), the distance until the next direction change along the the name of the street to take there and the total duration until the target. This is especially useful when I want to keep my phone in the pocket - for example when on my bike or walking in unknown areas where I don't want to carry my phone all the time in my hand.

However I prefer OSMAnd as it provides a much better navigation in many cases. I already asked the provider of the companion app if they are willing to extend the app for OSMAnd, since OSMAnd also displays navigation hints as notification and the companion app just reads the notification and forwards the information to the the smartwatch where a client app recieves it and displays it. However so far I did not get any response and I doubt I ever will :-(

### Tell us about the expected behaviour (required)

It would be nice to have some plugin in OSMAnd which can forward navigation information to a smartwatch app which will then display it - just a symbol for the direction, distance to next turn, total distance remaining and estimated time of arrival. There is no need for any map details.

Configurable options:

The plugin would automatically recognize when a navigation starts and contact the smartwatch app and will also tell the smartwatch app, when then navigation ends.

### Tell us about alternatives you've considered (required)

Create a companion app which will run as service in the background and catch all navigation notifications created by OSMAnd and forward them to a smartwatch app where they will be displayed.

### Context (optional)

See this app for reference: https://play.google.com/store/apps/details?id=smartwatchstudios.app.gears3navigation

I am also a software developer and can help if someone may point me to the relevant information how to get started.

**Comment by GHT23** [comment]
Would be very nice to have...

**Comment by tiburon-777** [comment]
Would be very nice to have...

**Comment by swdick2** [comment]
Yes, terrific idea. There seems to be something called Navismartwatch, but i have not the slightest idea how to get this and how to install and use it. Any suggestions?

**Comment by boredsquirrel** [comment]
I think GadgetBridge supports something in that area? It can connect to OSMAnd somehow (you can set the app name in the settings)

But I dont know what it does and if it works.

Also, this should be possible without touching any OSMAnd code at all, as you already said. OSMAnd should not take care of communication with the smartwatch. But it could connect to the smartwatch app (Gadgetbridge, Pebble, something proprietary) to do way more than just turn directions.

It could actually display a minimap of the area! While a BLE connection might not be fast enough to stream lots of images, maybe some kind of simplified vector format, together with an app e.g. on a Pebble.
